rs/replicated_state/src/replicated_state.rs

```diff
@@ -731,6 +732,26 @@ impl ReplicatedState {
             .sum()
     }
 
+    /// Canister migrations require that a canister is stopped, has no guaranteed responses
+    /// in any outgoing stream, and nothing in the output queue (guaranteed or otherwise).
+    pub fn ready_for_migration(&self, canister: &CanisterId) -> bool {
+        let streams_flushed = || {
+            self.metadata
+                .streams
+                .iter()
+                .all(|(_, stream)| stream.guaranteed_responses_counts().get(canister).is_none())
+        };
```

Comment:
If we're going to resurrect the per-`Streams` stats for `O(1)` "ready for migration" checks [to be included in the `canister_status()` output](https://dfinity.atlassian.net/browse/MR-665), do we also need this?

Reply:
I thought we had agreed to implement the `O(1)` solution. Or, at least, I remember pushing back against it since we'd just dropped the code; and ultimately giving in. (o:

Regardless, LGTM.